L/llvm_common.jl

```diff
@@ -0,0 +1,262 @@
+
+cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]}
+ninja -j${nproc} llvm-tblgen clang-tblgen llvm-config
+popd
+
+# Let's do the actual build within the `build` subdirectory
+mkdir ${WORKSPACE}/build && cd ${WORKSPACE}/build
+
+# Accumulate these flags outside CMAKE_FLAGS,
+# they will be added at the end.
+CMAKE_CPP_FLAGS=""
+CMAKE_CXX_FLAGS=""
+CMAKE_C_FLAGS=""
+
+CMAKE_FLAGS=()
+
+# Release build for best perofrmance
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
+if [[ "${ASSERTS}" == "1" ]]; then
+    CMAKE_FLAGS+=(-DLLVM_ENABLE_ASSERTIONS:BOOL=ON)
+fi
+
+# build for our host arch and our GPU targets NVidia and AMD
+TARGETS=(host NVPTX AMDGPU)
+# Add WASM for LLVM 8
+if [[ "${LLVM_MAJ_VER}" == "8" ]]; then
+    TARGETS+=(WebAssembly)
+fi
+LLVM_TARGETS=$(IFS=';' ; echo "${TARGETS[*]}")
+CMAKE_FLAGS+=(-DLLVM_TARGETS_TO_BUILD:STRING=$LLVM_TARGETS)
+
+# We mostly care about clang and LLVM
+CMAKE_FLAGS+=(-DLLVM_ENABLE_PROJECTS='clang;compiler-rt;lld')
+CMAKE_FLAGS+=(-DLLVM_TOOL_CLANG_TOOLS_EXTRA_BUILD=OFF)
+
+# We want a build with no bindings
+CMAKE_FLAGS+=(-DLLVM_BINDINGS_LIST="" )
+
+# Turn off ZLIB and XML2
+CMAKE_FLAGS+=(-DLLVM_ENABLE_ZLIB=OFF)
+CMAKE_FLAGS+=(-DLLVM_ENABLE_LIBXML2=OFF)
+
+# Disable useless things like docs, terminfo, etc....
+CMAKE_FLAGS+=(-DLLVM_INCLUDE_DOCS=Off)
+CMAKE_FLAGS+=(-DLLVM_ENABLE_TERMINFO=Off)
+CMAKE_FLAGS+=(-DHAVE_HISTEDIT_H=Off)
+CMAKE_FLAGS+=(-DHAVE_LIBEDIT=Off)
+
+# We want a shared library
+CMAKE_FLAGS+=(-DLLVM_BUILD_LLVM_DYLIB:BOOL=ON)
+CMAKE_FLAGS+=(-DLLVM_LINK_LLVM_DYLIB:BOOL=ON)
+
+# Install things into $prefix, and make sure it knows we're cross-compiling
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=True)
+
+# Julia expects the produced LLVM tools to be installed into tools and not bin
+# We can't simply move bin to tools since on MingW64 it will also contain the shlib.
+CMAKE_FLAGS+=(-DLLVM_TOOLS_INSTALL_DIR=${prefix}/tools)
+
+# Also build and install utils, since we want FileCheck, and lit
+CMAKE_FLAGS+=(-DLLVM_UTILS_INSTALL_DIR=${prefix}/tools)
+CMAKE_FLAGS+=(-DLLVM_INCLUDE_UTILS=True -DLLVM_INSTALL_UTILS=True)
+
+# Include perf/oprofile/vtune markers
+if [[ ${target} == *linux* ]]; then
+    CMAKE_FLAGS+=(-DUSE_PERF=1)
+    CMAKE_FLAGS+=(-DUSE_OPROFILE=1)
+fi
+if [[ ${target} == *linux* ]] || [[ ${target} == *mingw32* ]]; then
+    CMAKE_FLAGS+=(-DUSE_INTEL_JITEVENTS=1)
+fi
+
+# Tell LLVM where our pre-built tblgen tools are
+CMAKE_FLAGS+=(-DLLVM_TABLEGEN=${WORKSPACE}/bootstrap/bin/llvm-tblgen)
+CMAKE_FLAGS+=(-DCLANG_TABLEGEN=${WORKSPACE}/bootstrap/bin/clang-tblgen)
+CMAKE_FLAGS+=(-DLLVM_CONFIG_PATH=${WORKSPACE}/bootstrap/bin/llvm-config)
+
+# Explicitly use our cmake toolchain file
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=/opt/${target}/${target}_gcc.cmake)
+# XXX: Fixed by https://github.com/JuliaPackaging/Yggdrasil/pull/318
+if [[ "${target}" == *apple* ]]; then
+    sed -i "s/{{TARGET}/${target}/g" /opt/${target}/${target}_gcc.cmake
```

Comment:
Doesn't this make C++ compiler wrong?

Reply:
No, ok, I see what happened!

Reply:
It applies this fix: https://github.com/JuliaPackaging/Yggdrasil/pull/318/commits/f6246cd8a8a6a79f16128e9929b4bb4b1d96772b

Reply:
Yes, that was clear to me, but I was thinking that this would also transform `${{TARGET}}` into `${{TARGET}}}` in the line below. Instead what happened was that the template had `${{TARGET}` unexpanded on this line only.